mono/mini/intrinsics.c

```diff
@@ -399,8 +399,12 @@ emit_unsafe_intrinsics (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignatu
 			ins->klass = mono_get_object_class ();
 			return ins;
 		} else if (ctx->method_inst->type_argc == 1) {
-			if (mini_is_gsharedvt_variable_type (t))
+			if (mini_is_gsharedvt_variable_type (t)) {
+#ifdef ENABLE_NETCORE
+				g_warning ("Unable to intrinsify Unsafe.As<T>(object value) call in %s since T is a gsharedvt variable type: %s.", mono_method_full_name (cfg->method, 1), mono_type_full_name (t));
+#endif
```

Comment:
This is not needed.

Reply:
yeah not necessarily but since netcore Unsafe.As doesn't have a fallback you'll just get a PlatformNotSupportedException. I mainly put it here so we don't forget about it.

Reply:
We don't generally emit warnings in the JIT, the exception at runtime is enough to not forget it.